src/ceph-daemon/ceph-daemon

```diff
@@ -1925,6 +1928,11 @@ if __name__ == "__main__":
         logging.basicConfig(level=logging.INFO)
     logger = logging.getLogger('ceph-daemon')
 
+    # root?
+    if os.geteuid() != 0:
```

Comment:
Wouldn't this prevent running `rootless containers`  (e.g. run `ceph-deamon` as a non-root user `ceph`) ?

Reply:
Wont work that easy. This generates nasty errors as the systemctl commands change. Having a non-root version would be awesome, but this is bigger than just not checking for uid. Maybe adding a comment about other what needs to be done would be helpful.

Reply:
Right.. we can remove or change this if/when we support rootless containers.  They certainly don't work now